templates/terraform/resource.erb

```diff
@@ -169,22 +169,27 @@ func resource<%= resource_name -%>Update(d *schema.ResourceData, meta interface{
 	}
 
 	<% if object.input -%>
+	var obj map[string]interface{}
 	var url string
 	var res map[string]interface{}
 	op := &<%= api_name_lower -%>.Operation{}
 
 	d.Partial(true)
 
-	<% properties_by_custom_update(settable_properties).each do |key, props| -%>
+	<% properties_by_custom_update(properties).each do |key, props| -%>
 	if <%= props.map { |prop| "d.HasChange(\"#{Google::StringUtils.underscore(prop.name)}\")" }.join ' || ' -%> {
 		<% props.each do |prop| -%>
-			<%= prop.api_name -%>Prop, err := expand<%= resource_name -%><%= titlelize_property(prop) -%>(d.Get("<%= Google::StringUtils.underscore(prop.name) -%>"), d, config)
+      <% if settable_properties.include? prop -%>
```

Comment:
The indentation on this looks weird, but it's not because of your change- looks like some of the surrounding code is using tabs. Mind fixing that while you're here?

Reply:
Almost all this code uses tabs.  Converted what didn't use tabs to use tabs.